engine/src/main/java/org/terasology/reflection/copy/strategy/SetCopyStrategy.java

```diff
@@ -32,11 +33,7 @@ public SetCopyStrategy(CopyStrategy<T> contentStrategy) {
     @Override
     public Set<T> copy(Set<T> value) {
         if (value != null) {
-            Set<T> result = Sets.newHashSetWithExpectedSize(value.size());
-            for (T item : value) {
-                result.add(contentStrategy.copy(item));
-            }
-            return result;
+            return value.stream().map(contentStrategy::copy).collect(Collectors.toCollection(HashSet::new));
```

Comment:
I read the javadoc and found that Java specification doesn't enforce the type of the set.
Though the default now is a HashSet: I can use toSet if you want.